pallets/adz/src/lib.rs

```diff
@@ -12,26 +24,32 @@ mod tests;
 #[cfg(feature = "runtime-benchmarks")]
 mod benchmarking;
 
-use sp_std::collections::{btree_map::*, btree_set::*};
-use sp_std::prelude::Vec;
+pub use pallet::*;
+use pallet_timestamp as timestamp;
+
+#[derive(Encode, Decode, PartialEq, RuntimeDebug, scale_info::TypeInfo)]
+pub struct Ad<AccountId> {
+    pub author: AccountId,
+    pub selected_applicant: Option<AccountId>,
+    pub title: Vec<u8>,
+    pub body: Vec<u8>,
+    pub tags: Vec<Vec<u8>>,
+    pub created: u64,
+    pub num_of_comments: u32,
+}
+
+#[derive(Encode, Decode, PartialEq, sp_runtime::RuntimeDebug, scale_info::TypeInfo)]
+pub struct Comment<AccountId> {
+    author: AccountId,
+    body: Vec<u8>,
+    created: u64,
+}
```

Comment:
I needed to move this types out of the module below in order for the `TypeInfo` derive to work.